src/main/java/seedu/address/model/TaskBook.java

```diff
@@ -119,6 +142,17 @@ public EventTask removeEventTask(int index) {
         return eventTasks.remove(index);
     }
 
+    /**
+     * Remove the EventTasks that are filtered by the given predicate.
+     */
+    public Collection<EventTask> removeEventTask(Predicate<EventTask> predicate) {
+        Collection<EventTask> toRemove = FXCollections.observableArrayList();
```

Comment:
Why is an observable array is required?

Given that you return it as a `Collection<EventTask>`, I don't see the point.